x-pack/plugins/security_solution/public/common/components/search_bar/index.tsx

```diff
@@ -312,12 +321,12 @@ export const SearchBarComponent = memo<SiemSearchBarProps & PropsFromRedux>(
       },
       [filterManager, id, setTablesActivePageToZero, updateSearch]
     );
-    return (
+    return dataViews ? (
       <div data-test-subj={`${id}DatePicker`}>
         <SearchBar
           appName="siem"
           isLoading={isLoading}
-          indexPatterns={indexPatterns as DataView[]}
+          indexPatterns={dataViews}
```

Comment:
Remove the type casting here by implementing Line 299-305, so we can get the exact `DataView` type. This allows Search Bar to receive the correct data view and fix https://github.com/elastic/kibana/issues/164406